<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <title>Title</title>
</head>
<body>
    <div>
        <button>get</button>
        <button>post</button>
        <button>put</button>
        <button>delete</button>
    </div>

    <script src="axios.js"></script>
    <script>


        // Add a request interceptor
        axios.interceptors.request.use(function (config) {
            console.log('请求拦截器-1 成功')
            // Do something before request is sent
            return config;
        }, function (error) {
            console.log('请求拦截器-1 失败')
            // Do something with request error
            return Promise.reject(error);
        });

        axios.interceptors.request.use(function (config) {
            console.log('请求拦截器-2 成功')
            // Do something before request is sent
            return config;
        }, function (error) {
            console.log('请求拦截器-2 失败')
            // Do something with request error
            return Promise.reject(error);
        });

        // Add a response interceptor
        axios.interceptors.response.use(function (response) {
            console.log('响应拦截器-1 成功')
            // Any status code that lie within the range of 2xx cause this function to trigger
            // Do something with response data
            return response;
        }, function (error) {
            console.log('响应拦截器-1 失败')
            // Any status codes that falls outside the range of 2xx cause this function to trigger
            // Do something with response error
            return Promise.reject(error);
        });

        // Add a response interceptor
        axios.interceptors.response.use(function (response) {
            console.log('响应拦截器-2 成功')
            // Any status code that lie within the range of 2xx cause this function to trigger
            // Do something with response data
            return response;
        }, function (error) {
            console.log('响应拦截器-2 失败')
            // Any status codes that falls outside the range of 2xx cause this function to trigger
            // Do something with response error
            return Promise.reject(error);
        });

        axios({
            method: 'get',
            url: 'http://localhost:3000/posts/2',
        }).then(response => {
            console.log(response)
        })
    </script>
</body>
</html>


















